lib/rucio/core/authentication.py

```diff
@@ -223,7 +223,9 @@ def get_auth_token_ssh(account, signature, appid, ip=None, session=None):
     # try all available SSH identities for the account with the provided signature
     match = False
     for identity in identities:
-        pub_k = paramiko.RSAKey(data=b64decode(identity['identity'].split()[1]))
+        data = identity['identity'].split()[1]
+        data += '=' * ((4 - len(data) % 4) % 4)  # adding required padding
```

Comment:
There is no need to apply modulo twice: `'=' * ((4 - len(data)) % 4)`

And since you're using the Python specific modulo for negative values, please add a comment explaining it. e. g. `-1 % 4 = 3` – I don't think that's very intuitive ;)

this comment applies to all other similarly changed lines as well

Reply:
now that I think about it, it's fine as is :) Sorry for the confusion